File: app/src/main/java/com/chinachino/mvvm/Scheduler.java
package com.chinachino.mvvm;

import androidx.lifecycle.MutableLiveData;

import com.chinachino.mvvm.Executors.AppExecutors;

import java.util.concurrent.Future;
import java.util.concurrent.TimeUnit;

import static com.chinachino.mvvm.Utils.Constants.TIME_OUT;

public class Scheduler {
    public void Schedule(Future future, MutableLiveData<Boolean> RequestTimeOut){
        AppExecutors.getInstance().getExecutorService().schedule(() -> {
            RequestTimeOut.postValue(true);
            future.cancel(true);
        }, TIME_OUT, TimeUnit.SECONDS);
    }
}

File: app/src/main/java/com/chinachino/mvvm/requests/Scheduler.kt
package com.chinachino.mvvm.requests

import androidx.lifecycle.MutableLiveData
import com.chinachino.mvvm.utils.Constants
import com.chinachino.mvvm.executors.AppExecutors
import java.util.concurrent.Future
import java.util.concurrent.TimeUnit

class Scheduler {
    fun schedule(future: Future<*>, RequestTimeOut: MutableLiveData<Boolean?>) {
        AppExecutors.executorService.schedule({
            RequestTimeOut.postValue(true)
            future.cancel(true)
        }, Constants.TIME_OUT, TimeUnit.SECONDS)
    }
}